enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,428 @@
+     according to the HostSubnet CR of this node.
+
+#### Migration
+
+When the condition 'Progressing' of the network Cluster Operator is switched
+from `True` to `False`, we can kick off the migration from CNO, by:
+
+```bash
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"spec":{"networkType":"OVNKubernetes"}'
+```
+
+CNO will update the `status.networkType` field of the `network.config` CR. When
+MCO renders the MachineConfig, some host network configuration service will be
+rendered differently for running different CNI. For example, the
+`ovs-configuration.service` will create the br-ex bridge if the
+`status.networkType` is `OVNKubernetes`, it will not otherwise. So the change of
+the `status.networkType` field will trigger MCO to apply new MachineConfig to
+each node:
+
+1. MCO will 
+   - rerender the MachineConfig for each node
+   - try to cordon and drain the nodes based on
+     `MachineConfigPools.spec.maxUnavailable`
+2. MCO will drain and reboot the node.
+3. When the node is up, br-ex will be created. it means the node is ready for
+   OVN Kubernetes. So the local zone OVN Kubernetes components can be started.
+   The ovnkube-node container wrapper script will also add the node annotation
+  `k8s.ovn.org/node-subnets`, according to the HostSubnet CR of this node.
+4. On local node, OVN Kubernetes will add all SDN nodes as hybrid overlay nodes
+   to local zone, and all OVN nodes as OVN-IC remote zones. On remote OVN node,
+   OVN Kubernetes will remove a hybrid overlay node and add a OVN-IC remote
+   zone.
+5. MCO will uncordon the node. Pods will be recreated on the node using
+   ovn-kubernetes as the default CNI plugin.
+
+The above process will be repeated for each node, until all the nodes have been
+applied to the new MachineConfig and converted to OVN Kubernetes.
+
+#### Migration Cleanup
+
+Once migration is complete, CNO will:
+- delete the openshift-sdn DaemonSets and the related resources
+  (CustomResources, ConfigMaps, etc).
+- redeploy ovn-kubernetes in "normal" mode (no migration mode script, hybrid
+  overlay disabled, etc).
+- remove the migration-related labels and node annotations from the nodes.
+      // live indicates whether we're doing a live migration or not.
+
+### API
+
+A new field `spec.migration.live` will be introduced to the CRD
+`networks.operator.openshift.io`. Users can set this flag to conduct a live
+migration or a offline migration.
+
+To start the migration, users need to update the `network.operator`
+CR by adding:
+
+CNO will deploy both OpenShift SDN and OVN Kubernetes pods in 'migration' mode.
+
+```json
+{ 
+  "spec": { 
+    "migration": {
+      "networkType": "OVNKubernetes",
+      // live indicates whether we're doing a live migration or not.
+      "live": true
+    }
+  } 
+}
+```
+
+On removing of the `spec.migration` field, CNO will start the migration cleanup
+and redeploy the target CNI in 'normal' mode.
+
+```
```

Comment:
this causes the entire rest of the doc to be in a code block

Reply:
Fixed.